contracts/BoostedMasterChefJoe.sol

```diff
@@ -0,0 +1,621 @@
+    /// @param dummyToken The address of the ERC-20 token to deposit into MCJV2.
+    function init(IERC20 dummyToken) external onlyOwner {
+        uint256 balance = dummyToken.balanceOf(msg.sender);
+        require(balance != 0, "MasterChefV2: Balance must exceed 0");
+        dummyToken.safeTransferFrom(msg.sender, address(this), balance);
+        dummyToken.approve(address(MASTER_CHEF_V2), balance);
+        MASTER_CHEF_V2.deposit(MASTER_PID, balance);
+        emit Init();
+    }
+
+    /// @notice Returns the number of BMCJ pools.
+    function poolLength() external view returns (uint256 pools) {
+        pools = poolInfo.length;
+    }
+
+    /// @notice Add a new LP to the pool. Can only be called by the owner.
+    /// DO NOT add the same LP token more than once. Rewards will be messed up if you do.
+    /// @param allocPoint AP of the new pool.
+    /// @param _lpToken Address of the LP ERC-20 token.
+    /// @param _rewarder Address of the rewarder delegate.
+    function add(
+        uint256 allocPoint,
+        IERC20 _lpToken,
+        IRewarder _rewarder
+    ) external onlyOwner {
+        require(!lpTokens.contains(address(_lpToken)), "add: LP already added");
+        require(poolInfo.length < 50, "Too many pools");
+        // Sanity check to ensure _lpToken is an ERC20 token
+        _lpToken.balanceOf(address(this));
+        // Sanity check if we add a rewarder
+        if (address(_rewarder) != address(0)) {
+            _rewarder.onJoeReward(address(0), 0);
+        }
+
+        uint256 lastRewardTimestamp = block.timestamp;
+        totalAllocPoint = totalAllocPoint.add(allocPoint);
+
+        poolInfo.push(
+            PoolInfo({
+                lpToken: _lpToken,
+                allocPoint: allocPoint,
+                lastRewardTimestamp: lastRewardTimestamp,
+                accJoePerShare: 0,
+                rewarder: _rewarder,
+                totalVeJoe: 0,
+                totalBoostedAmount: 0
+            })
+        );
+        lpTokens.add(address(_lpToken));
+        emit Add(poolInfo.length.sub(1), allocPoint, _lpToken, _rewarder);
+    }
+
+    /// @notice Update the given pool's JOE allocation point and `IRewarder` contract. Can only be called by the owner.
+    /// @param _pid The index of the pool. See `poolInfo`.
+    /// @param _allocPoint New AP of the pool.
+    /// @param _rewarder Address of the rewarder delegate.
+    /// @param overwrite True if _rewarder should be `set`. Otherwise `_rewarder` is ignored.
+    function set(
+        uint256 _pid,
+        uint256 _allocPoint,
+        IRewarder _rewarder,
+        bool overwrite
+    ) external onlyOwner {
+        PoolInfo memory pool = poolInfo[_pid];
+        totalAllocPoint = totalAllocPoint.sub(poolInfo[_pid].allocPoint).add(_allocPoint);
+        pool.allocPoint = _allocPoint;
+        if (overwrite) {
+            _rewarder.onJoeReward(address(0), 0); // sanity check
+            pool.rewarder = _rewarder;
+        }
```

Comment:
```
        if (overwrite) {
           if (_rewarder != address(0) {
                _rewarder.onJoeReward(address(0), 0); // sanity check
            }
            pool.rewarder = _rewarder;
        }
 ```

Reply:
I'm going to leave this change until later, this code is vebatim from MCV3 and I'm hesitant to change it.